course_info/static/course_info/js/controllers/SearchController.js

```diff
@@ -57,7 +57,8 @@
                                 text: tc.term_name + ' <span class="caret"></span>',
                             };
                         }));
-                    if (response.data.next !== '') {
+                    if (response.data.next && response.data.next !== '') {
+                        // API returns next=null if there are no more pages
```

Comment:
Hah...that's been annoying me for months.  Thanks for hunting it down, and especially for including the comment explaining it.